src/lib.rs

```diff
@@ -1,8 +1,254 @@
+use anyhow::{Context, Result};
+use clap::Parser;
+use lazy_static::lazy_static;
+use notify::{RecommendedWatcher, RecursiveMode, Watcher};
+use std::{
+    path::{Path, PathBuf},
+    process,
+    sync::mpsc,
+    time::{Duration, Instant},
+};
+
+pub use anyhow;
+pub use cargo_metadata;
+pub use cargo_metadata::camino;
+pub use clap;
+
+/// Fetch the metadata of the crate.
+pub fn metadata() -> &'static cargo_metadata::Metadata {
+    lazy_static! {
+        static ref METADATA: cargo_metadata::Metadata = cargo_metadata::MetadataCommand::new()
+            .exec()
+            .expect("cannot get crate's metadata");
+    }
+
+    &METADATA
+}
+
+/// Fetch information of a package in the current crate.
+pub fn package(name: &str) -> Option<&cargo_metadata::Package> {
+    metadata().packages.iter().find(|x| x.name == name)
+}
+
+/// Watches over your project's source code, relaunching the given command when
+/// changes are detected.
+#[non_exhaustive]
+#[derive(Debug, Parser)]
+pub struct Watch {
+    /// Watch specific file(s) or folder(s). The default is the workspace root.
+    #[clap(long = "watch", short = 'w')]
+    pub watch_paths: Vec<PathBuf>,
+    /// Paths that will be excluded.
+    #[clap(long = "ignore", short = 'i')]
+    pub exclude_paths: Vec<PathBuf>,
+    /// Paths, relative to the workspace root, that will be excluded.
+    #[clap(skip)]
+    pub workspace_exclude_paths: Vec<PathBuf>,
+    /// Set the debounce duration after relaunching a command.
+    ///
+    /// The default is 2 seconds.
+    #[clap(skip)]
+    pub debounce: Option<Duration>,
```

Comment:
```suggestion
    /// Throttle events to prevent the command to be re-executed too early right after an execution already occurred.
    ///
    /// The default is 2 seconds.
    #[clap(skip = Duration::from_secs(2))]
    pub debounce: Duration,
```